pkg/plugin/loader.go

```diff
@@ -82,14 +87,59 @@ func checkDirAndCompile(pluginType, driver string) (string, error) {
 	return bin, nil
 }
 
+// downloadPlugin downloads the plugin and stores it into local filesystem
+func downloadAndCompilePlugin(pluginType, driver string) (string, error) {
+	driverURL := strings.TrimPrefix(driver, "https://")
+	destination := fmt.Sprintf("%s/ext/%s/%s", dirname, pluginType, filepath.Base(driver))
+	client := &getter.Client{
+		Ctx:  context.Background(),
+		Dst:  destination,
+		Src:  driverURL,
+		Mode: getter.ClientModeDir,
+	}
+	if err := client.Get(); err != nil {
+		return "", err
+	}
+	bin, err := compile(pluginType, destination)
+	if err != nil {
+		return "", nil
+	}
+	return bin, nil
+}
+
+// isValidURL tests a string to determine if it is a well-structure URL
+func isValidURL(driver string) bool {
+	_, err := url.ParseRequestURI(driver)
+	if err != nil {
+		return false
+	}
+
+	u, err := url.Parse(driver)
+	if err != nil || u.Scheme == "" || u.Host == "" {
+		return false
+	}
+
+	return true
+}
+
 // Load loads the plugin using the hashicorp go-plugin system
 func Load(pluginType, driver string) (*RevaPlugin, error) {
+	var bin string
+	var err error
 	if isAlphaNum(driver) {
```

Comment:
The in-memory plugins can have hyphens and underscores, so let's add that to the regex